### PR TITLE
Initial FreeBSD support

### PR DESCRIPTION
**Pull request by arrowd**
This PR contains commits that enable building pipewire on FreeBSD OS. I didn't test it extensively, but at least some executables seem to work.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -207,6 +207,10 @@ if get_option('gstreamer')
              dependency('gstreamer-allocators-1.0'),]
 endif
 
+if build_machine.system() == 'freebsd'
+  epoll_shim_dep = dependency('epoll-shim', required: true)
+endif
+
```
Comment (Hi-Angel) [inline meson.build]: You leave this undefined unless the platform is freebsd. I.e. if you try to compile it on any other, it gonna break the build. You probably want something like this instead:

```meson
epoll_shim_dep = (build_machine.system() == 'freebsd') \
  ? dependency('epoll-shim', required: true) : dependency('', required: false)
```

**Comment by Hi-Angel** [comment]
Not a project member, but something I noticed from a cursory look: the comment above and also that your commits have first line too long. If you try to use `git shortlog`, etc, it gonna look ugly. The first line should have brief description of what the commit does, then empty line, then optionally more info about the commit.

File: meson.build (hunk 1 above)
Comment (Hi-Angel) [reply meson.build]: I also think you may want to move the description about epoll-shiim that you have in commit message here in comment instead.

**Comment by arrowd** [comment]
Addressed comments and simplified meson changes.

**Comment by arrowd** [comment]
Ping.

**Comment by wtay** [comment]
hi, thanks for the patch, I'll have a look at it soon (and we're actually moved to gitlab on freedesktop)

**Comment by arrowd** [comment]
> hi, thanks for the patch, I'll have a look at it soon

Great, thanks!

> (and we're actually moved to gitlab on freedesktop)

I saw that, but got impression that PRs on github are also get accepted.

**Comment by wtay** [comment]
pushed to the gitlab repo.

In theory, all the eventfd/signalfd/timerfd stuff should be done through the spa_system interface, which is pluggable, and the only place to add the epoll_shim dependency should be there.

I fixed all cases where it was not using the system library, please retest.

**Comment by arrowd** [comment]
> pushed to the gitlab repo.

Great, thanks!

> 
> In theory, all the eventfd/signalfd/timerfd stuff should be done through the spa_system interface, which is pluggable, and the only place to add the epoll_shim dependency should be there.
> 
> I fixed all cases where it was not using the system library, please retest.

It doesn't compile now. Many targets fail due to following error:

```
../spa/include/spa/support/system.h:32:10: fatal error: 'sys/timerfd.h' file not found
#include <sys/timerfd.h>
         ^~~~~~~~~~~~~~~
```

Another patch I had to apply:

```
diff --git a/spa/plugins/support/logger.c b/spa/plugins/support/logger.c
index f9197a56..1b9d1079 100644
--- a/spa/plugins/support/logger.c
+++ b/spa/plugins/support/logger.c
@@ -37,6 +37,10 @@
 #include <spa/utils/type.h>
 #include <spa/utils/names.h>
 
+#ifdef __FreeBSD__
+#define CLOCK_MONOTONIC_RAW CLOCK_MONOTONIC
+#endif
+
 #define NAME "logger"
 
 #define DEFAULT_LOG_LEVEL SPA_LOG_LEVEL_INFO
```

**Comment by wtay** [comment]
#include <sys/timerfd.h>

This is just for the itimerspec struct definition, which is also in <time.h>
If I replace this with:

#include <time.h>

it still works on linux. I've pushed this and the logger fix.

**Comment by arrowd** [comment]
I confirm it builds now. Thanks for cooperation!

Would you be interested in adding FreeBSD job to GitLab CI? Docker doesn't support FreeBSD, but GitLab also has VirtualBox executor, which can be used to run builds inside FreeBSD VM.

**Pull request closed without merging**